### PR TITLE
docs: Full Translation to Spanish

### PR DESCRIPTION
**Pull request by jefer590**
Add Full Spanish Translation to the docs

### Note
This will need a couple of revisions before releasing this.

### CODE REVIEW

File: docs/es/guide/setup.md [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+:::
+
+::: tip
+
+Checa la [documentación oficial de TypeScript](https://www.typescriptlang.org/docs/handbook/compiler-options.html) para aprender sobre las diferentes opciones del compilador.
+:::
+
+Y es todo! Ya estás listo para usar TypeScript en tus **layouts**, **components**, **plugins** y **middlewares**.
+
+Puedes tambien checar la sección de [**Recetas de Cocina**](../cookbook/components/) para obtener algunas recetas usando TypeScript para tu proyecto con Nuxt.
```
Comment (marconi1992) [inline docs/es/guide/setup.md]: Podríamos usar otro verbo sinónimo de `checar` como `revisar`, `mirar`, `examinar` . En mi opinión `mirar` suena mejor.

**Comment by MrJmpl3** [comment]
Si gustan pueden mirar mi traducción y hacer una comparación .. https://github.com/nuxt/typescript/pull/178

File: docs/es/guide/setup.md (hunk 1 above)
Comment (MrJmpl3) [reply docs/es/guide/setup.md]: A mi me gusta , `revisar`

Y veo que han traducido `CookBook` , creo que esa palabra se entiende mejor en Ingles, porque en Español, como que pierde cierto sentido

Comment (jackbravo) [reply docs/es/guide/setup.md]: Para la gente que sabe algo de inglés, cookbok se entiende mejor, pero no puedes saber si eso será cierto de las personas que lean las traducciones. Justo lo puede leer alguien que sabe poco o nada de inglés. A lo mejor yo lo cambiaría por `Recetario` que es más corto. Sólo no se que tan usado sea fuera de México, que también la traducción debe buscar ser de un español neutro para evitar tener traducciones en español de España separadas de las de México como en Firefox 😭.

**Comment by kevinmarrec** [comment]
@MrJmpl3 Is your work somehow has been merged into this one ?

**Comment by kevinmarrec** [comment]
@jefer590 I'm closing https://github.com/nuxt/typescript/issues/190 but please take care looking at it if there's something relevant about spanish translation

**Comment by MrJmpl3** [comment]
@kevinmarrec No, this is another fork for translate to spanish

Merge the any of the two PR to make a base to make more PR to fixes some typos

**Comment by kevinmarrec** [comment]
@MrJmpl3 Well I need someone to tell me which one is the best "base", so I can close the other (or let it be updated afterward), both seem to have almost same number of files changed

**Comment by kevinmarrec** [comment]
Maybe I can create a spanish translation branch and merge one of them so you can continue contribute on the translation before we officialy release a finished full spanish version of the documentation ?

**Comment by MrJmpl3** [comment]
Create a branch with this PR , I will a check the translation file per file ..

**Comment by Blask0** [comment]
Hi, how can I help you to translate the page?

File: docs/es/index.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+heroImage: /assets/logo.svg
+actionText: Empieza Ahora →
+actionLink: /guide/
+footer: MIT Licensed | Copyright © 2019-present Nuxt.js Team
+---
+
+<div class="features">
+  <div class="feature">
+    <h2>Autocompletado de código</h2>
+    <p>Una de las mas grandes ventahjas de TypeScript es su autocompletado de código e IntelliSense.</p>
```
Comment (Blask0) [inline docs/es/index.md]: Una de las mas grandes ventahjas // Una de las ventajas más grandes

File: docs/es/index.md [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+  </div>
+  <div class="feature">
+    <h2>Seguridad de Tipos</h2>
+    <p>Tipos incrementan tu agilidad al momento de hacer "refactors".</p>
+    <p> Es mejor que el compilador atrape errores que las cosas fallen en tiempo de ejecución.</p>
+  </div>
+  <div class="feature">
+    <h2>Calidad de Código</h2>
+    <p>Tipos tienen una demostrada habilidad de mejorar la calidad y la entendibilidad del código.</p>
+    <p>Grandes equipos (Google, Microsoft, Facebook) han continuamente llegado a esta conclusión.</p>
```
Comment (Blask0) [inline docs/es/index.md]: han continuamente llegado a esta conclusión // han llegado continuamente a esta conclusión

File: docs/es/index.md [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+<div class="features">
+  <div class="feature">
+    <h2>Autocompletado de código</h2>
+    <p>Una de las mas grandes ventahjas de TypeScript es su autocompletado de código e IntelliSense.</p>
+    <p> IntelliSense provee pistas activas en cuanto el código es agregado.</p>
+  </div>
+  <div class="feature">
+    <h2>Seguridad de Tipos</h2>
+    <p>Tipos incrementan tu agilidad al momento de hacer "refactors".</p>
+    <p> Es mejor que el compilador atrape errores que las cosas fallen en tiempo de ejecución.</p>
```
Comment (Blask0) [inline docs/es/index.md]: Es mejor que el compilador detecte errores antes de que las cosas fallen en el tiempo de ejecución.

File: docs/es/guide/setup.md [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+> Habilita la repreción de advertencias de TypeScript de no encontrado.
+
+- Tipo: `Boolean`
+- Predeterminado: `false`
+
+Cuando está habilotado, puedes reprimir las advertencias de tipo `export ... was not found ...`
+
+Vea también sobre información de antecedentes [aquí](https://github.com/TypeStrong/ts-loader/issues/653).
+
+**Advertencia:** Esta propiedad podría reprimir las advertencias que quieres ver. Ten cuidaddo en como lo configuras.
```
Comment (Blask0) [inline docs/es/guide/setup.md]: **Advertencia:** Esta propiedad podría reprimir las advertencias que quieres ver. Ten cuidado al configurarlo.

File: docs/es/guide/setup.md [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+Puedes usar un `Object` para sobreescribir opciones del plugin o ponerlo en `false` para deshabilitarlo.
+
+### ignoreNotFoundWarnings
+
+> Habilita la repreción de advertencias de TypeScript de no encontrado.
+
+- Tipo: `Boolean`
+- Predeterminado: `false`
+
+Cuando está habilotado, puedes reprimir las advertencias de tipo `export ... was not found ...`
```
Comment (Blask0) [inline docs/es/guide/setup.md]: "Habilitado"

**Comment by kevinmarrec** [comment]
@jefer590 I created a `spanish-translation` branch and edited this PR to target it.
I would like you take account of @Blask0 suggested changes if relevant before I merge it :)

**Comment by kevinmarrec** [comment]
I merged @MrJmpl3 PR into the branch as it was the oldest PR sitting here.
It needs to be polished through this one now, or through a new one if it's easier.

**Comment by MrJmpl3** [comment]
@kevinmarrec Thanks , i will a compare this PR to fix the branch.

**Comment by jefer590** [comment]
@kevinmarrec I will leverage the content of this PR taking into account the content of the made branch and all the suggestions by the people 🙂

**Comment by MrJmpl3** [comment]
I tried to look the diff of `nuxt:spanish-translation` and `jefer590:personal/jefer590/translation-to-es`

But, Github show the diff of `nuxt:master` and `jefer590:personal/jefer590/translation-to-es` 

Is a bug of Github? Or Am I wrong?

**Comment by kevinmarrec** [comment]
@MrJmpl3 It's probably cause of conflicts I'd say, so the diff doesn't have been properly updated i guess

**Comment by kevinmarrec** [comment]
Also @jefer590 branch needs to be rebased to have the goof diff

**Comment by jefer590** [comment]
@MrJmpl3 @kevinmarrec Branch is up to date 🙂

File: docs/es/cookbook/components/index.md [hunk 7]
```diff
@@ -1,9 +1,9 @@
-# Componentes
+# Components
```
Comment (kevinmarrec) [inline docs/es/cookbook/components/index.md]: Why is it translated back to english ?

Comment (jefer590) [reply docs/es/cookbook/components/index.md]: merge shenanigans 🙃 if you found more of this let me know to change it

File: docs/es/cookbook/components/index.md [hunk 8]
```diff
@@ -1,9 +1,9 @@
 # Componentes
```
Comment (kevinmarrec) [inline docs/es/cookbook/components/index.md]: Well I meant the whole page :) Seems to be the only one though

Comment (jefer590) [reply docs/es/cookbook/components/index.md]: i already pushed the fix for that 😄

Comment (kevinmarrec) [reply docs/es/cookbook/components/index.md]: Well AFAIK you only rollbacked the first line of the file 😬

**Comment by kevinmarrec** [comment]
Gonna take the lead on the fix with a new branch, so we finish this quicker

**Pull request merged**